src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -331,6 +345,20 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
     private fun pluralise(count: Int, singular: String, plural: String): String =
         if (count == 1) singular else plural
 
+    private fun canElideReturnLifetime(fn: PsiElement): Boolean {
```

Comment:
Hm, not sure that we already can handle this properly: the lifetime might be hidden inside a generic parameter:

```
struct S<'a> {
    foo: &'a u32,
}

fn main() {
    let x = 92;
    let s = S { foo: &x };
    let y = get_ref(s);
    println!("y = {:?}", y);
}

fn get_ref(s: S) -> &u32 {
    s.foo
}
```